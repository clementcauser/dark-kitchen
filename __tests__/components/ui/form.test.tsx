import { render } from "@testing-library/react";
import {
  FormContext,
  FormControl,
  FormDescription,
  FormField,
  FormItem,
  FormLabel,
  FormMessage,
} from "@/components/ui/form";
import { useForm } from "react-hook-form";

const Component = ({ name }: { name: string }) => {
  const form = useForm<{ name: string }>({
    defaultValues: { name },
  });

  return (
    <FormContext {...form}>
      <form>
        <FormField
          control={form.control}
          name="name"
          render={({ field }) => (
            <FormItem>
              <FormLabel>Nom</FormLabel>
              <FormControl>
                <input {...field} />
              </FormControl>
              <FormDescription>Votre nom de famille</FormDescription>
              <FormMessage />
            </FormItem>
          )}
        />
      </form>
    </FormContext>
  );
};

describe("form", () => {
  it("renders correctly", () => {
    const { container } = render(<Component name="default" />);

    expect(container).toMatchSnapshot();
  });
});
