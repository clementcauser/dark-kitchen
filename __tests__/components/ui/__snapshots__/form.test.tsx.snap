// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form renders correctly 1`] = `
<div>
  <form>
    <div
      class="space-y-1"
    >
      <label
        class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
        for=":r0:-form-item"
      >
        Nom
      </label>
      <input
        aria-describedby=":r0:-form-item-description"
        aria-invalid="false"
        id=":r0:-form-item"
        name="name"
        value="default"
      />
      <p
        class="text-xs text-muted-foreground"
        id=":r0:-form-item-description"
      >
        Votre nom de famille
      </p>
    </div>
  </form>
</div>
`;
